test/unity/toolkits/object_detection/test_od_data_iterator.cxx

```diff
@@ -119,6 +123,146 @@ BOOST_AUTO_TEST_CASE(test_simple_data_iterator) {
+        std::cout << "image : bottom_left" <<std::endl;
+        result.image_origin = data_iterator::image_origin_enum::BOTTOM_LEFT;
+    }
+    else {
+        // Add error logging
+    } 
+
+    if (annotation_scale == "pixel") {
+        std::cout << "annon scale: pixel" <<std::endl;
+        result.annotation_scale = data_iterator::annotation_scale_enum::PIXEL;
+    }
+    else if (annotation_scale == "normalized") {
+        std::cout << "annon scale: norm" <<std::endl;
+        result.annotation_scale = data_iterator::annotation_scale_enum::NORMALIZED;
+    }
+    else {
+        // Add error logging
+    }
+    
+    if (annotation_origin == "center") {
+        std::cout << "annon origin: center" <<std::endl;
+        result.annotation_origin = data_iterator::annotation_origin_enum::CENTER;
+    }
+    else if (annotation_origin == "top_left") {
+        std::cout << "annon origin: top_left" <<std::endl;
+        result.annotation_origin = data_iterator::annotation_origin_enum::TOP_LEFT;
+    }
+    else if (annotation_origin == "bottom_left") {
+        std::cout << "annon origin: bottom_left" <<std::endl;
+        result.annotation_origin = data_iterator::annotation_origin_enum::BOTTOM_LEFT;
+    }
+    else {
+        // Add error logging
+    }
+
+    // Each image has one annotation, with the label "foo" and a bounding box
+    // with height and width 16. As the row index increases, the box moves to
+    // the right until eventually resetting to the left and moving down.
+    annotations[0] = flex_list();
+    annotations[0].push_back(flex_dict({
+          {"label", "foo"},
+          {"coordinates", flex_dict({ {"x", x},
+                                      {"y", y},
+                                      {"width", w},
+                                      {"height", h}      })},
+        }));
+
+     
+    result.annotations_column_name = "test_annotations";
+    result.image_column_name = "test_image";
+    result.data =  gl_sframe({
+        {"test_image", gl_sarray(images)},
+        {"test_annotations", gl_sarray(annotations)},
+    });
+
+    result.shuffle = false;
+
+    simple_data_iterator data_source(result);
+    std::vector<labeled_image> batch = data_source.next_batch(1);
+    const labeled_image& example = batch[0];
+
+    return example;
+
+  };
+
+  // Case 1: Input: ('top_left', 'pixel', 'center') → ('top_left', 'normalized', 'top_left')
+  const labeled_image& case1_example = create_data_opts("top_left", "pixel", "center", 28, 18, 16, 16);
+  TS_ASSERT_EQUALS(case1_example.annotations[0].bounding_box,
+                      image_box(static_cast<float>(20) / IMAGE_WIDTH,
+                                 static_cast<float>(10) / IMAGE_HEIGHT,
+                                 16.f / IMAGE_WIDTH,
+                                 16.f / IMAGE_HEIGHT
+                                 ));
+
+  // Case 2: ('bottom_left', 'normalized', 'bottom_left') → ('top_left', 'normalized', 'top_left')
```

Comment:
Looks like this case is ‘pixel’ not ‘normalized’?

Reply:
Yes, typo. Thank you!